explainer-client/src/main/scala/ExplainEditorJS.scala

```diff
@@ -45,74 +45,118 @@ object ExplainEditorJS {
-      }
+    def publish(id: String): Future[Explainer] = {
+      Ajaxer[ExplainerApi].publish(id).call()
+    }
 
-      val indicatorId = s"presence-indicator-$area"
-      val fieldPresenceIndicator = div(`class` := "field-presence-indicator", id := indicatorId)()
+  }
 
-      presenceClient.on("visitor-list-updated", { data: js.Object =>
-        val stateChange = upickle.default.read[StateChange](js.JSON.stringify(data))
+  def presenceEnvelop(explainerId: String ,area: String, field: Element) = {
+    field.onfocus = (x: FocusEvent) => {
+      presenceClient.enter("explain-" + explainerId, area)
+    }
 
-        val statesOnThisArea: Seq[State] = stateChange.currentState.filter(_.location == area)
+    val indicatorId = s"presence-indicator-$area"
+    val fieldPresenceIndicator = div(`class` := "field-presence-indicator", id := indicatorId)()
 
-        dom.document.getElementById(indicatorId).innerHTML = statesOnThisArea.map(_.clientId.person.initials).mkString(" ")
-        ()
-      })
+    presenceClient.on("visitor-list-updated", { data: js.Object =>
+      val stateChange = upickle.default.read[StateChange](js.JSON.stringify(data))
+      val statesOnThisArea: Seq[State] = stateChange.currentState.filter(_.location == area)
+      dom.document.getElementById(indicatorId).innerHTML = statesOnThisArea.map(_.clientId.person.initials).mkString(" ")
+      ()
+    })
 
-      div(`class` := "presence-field-container") (
-        fieldPresenceIndicator,
+    div(`class` := "presence-field-container") (
+      fieldPresenceIndicator,
+      div(cls:="form-group")(
+        label(area),
         field
       )
+    )
+  }
+
+  def statusBar(explainer: Explainer) = {
+    val isDraftState =  !explainer.headline_published.isDefined || !explainer.body_published.isDefined || explainer.headline!=explainer.headline_published.get || explainer.body!=explainer.body_published.get
+    val status = if(isDraftState){
+      "Draft state: click on [Publish] to publish."
+    }else{
+      ""
     }
+    div(id:="explainer-editor__ops-env__status-bar-env__status-bar",cls:="red")(status)
+  }
+
+  def republishStatusBar(explainer: Explainer) = {
+    dom.document.getElementById("explainer-editor__ops-env__status-bar-env").innerHTML = statusBar(explainer).render.innerHTML
+  }
+
+  def ExplainEditor(explainerId: String, explainer: Explainer) = {
+
+    val headline: TypedTag[Input] = input(
+      id:="explainer-editor__headline-envelop__input",
+      cls:="explainer-input-field",
+      placeholder:="headline",
+      autofocus:=true
+    )
 
     val headlineTag = headline(value := explainer.headline).render
     headlineTag.onchange = (x: Event) => {
-      Model.saveContent(explainerId, ExplainerUpdate("headline", headlineTag.value))
-      false
+      Model.updateFieldContent(explainerId, ExplainerUpdate("headline", headlineTag.value)).map { explainer: Explainer =>
+        republishStatusBar(explainer)

```

Comment:
You could simplify this to 
`Model.updateFieldContent(explainerId, ExplainerUpdate("headline", headlineTag.value)).map(republishStatusBar)` - likewise below